### DISCUSSION TITLE
Word Wrap

### DISCUSSION THREAD

**Discussion by ghost** [PrismJS/prism#3360] [Q&A]
Hello, thanks for the amazing work.

is it possible to wrap to long words into next line just like in the VS code? This is useful if we have really long sentences/paragraphs because the long horizontal scrolling bar is really ugly...I've tried `word-break: break-all;` but it's not working.

I don't see any plugin can achieve this and I also seeing all the long horizontal scrolling bar in the demos of your official website too.

**Comment by RunDevelopment** [comment]
The CSS property you need is `white-space: pre-wrap;`. This will do the word wrapping. For more information, see [the MDN page](https://developer.mozilla.org/en-US/docs/Web/CSS/white-space).

**Reply by ghost** [reply]
> The CSS property you need is `white-space: pre-wrap;`. This will do the word wrapping. For more information, see [the MDN page](https://developer.mozilla.org/en-US/docs/Web/CSS/white-space).

It doesn't work, please try yourself...Thank you

**Reply by RunDevelopment** [reply]
Apply this property to both the `pre` and `code` element and it should work. For a working example see our [Line Numbers plugin page](https://prismjs.com/plugins/line-numbers/). 

Could you please provide an example for it not working? There might be something else interfering.

**Reply by ghost** [reply]
> Apply this property to both the `pre` and `code` element and it should work. For a working example see our [Line Numbers plugin page](https://prismjs.com/plugins/line-numbers/).
> 
> Could you please provide an example for it not working? There might be something else interfering.

I see, thanks. But it's not the ideal resolution as the second line will start without space. I mean we usually insert a tap space for child elements, but using `word-break: break-all;` just wrapping the sentences to the next line without following the tap space of the first line.

**Reply by RunDevelopment** [reply]
Unfortunately, indented code wraps are not supported by today's browsers. Implementing them on a library-level isn't easy either, so there hasn't been much progress toward that goal. For more information, see #2202.
